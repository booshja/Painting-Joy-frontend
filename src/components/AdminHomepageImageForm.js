// dependencies
import React from "react";
import styled from "styled-components";
// components
import { StyledP } from "../containers/styles/adminTypography";
import {
    StyledGreenSoloButton,
    StyledOutlineButton,
} from "../containers/styles/adminButtons";
// hooks
import { useForm } from "react-hook-form";
import { useHistory } from "react-router-dom";

const StyledUploadForm = styled.form`
    display: flex;
    flex-direction: column;
    border: 2px solid #207070;
    border-radius: 4px;
    background-color: #ffffff;
    padding: 1rem;
    width: 100%;
    margin-bottom: 1rem;
`;

const StyledUploadInput = styled.input`
    font-family: "News Cycle", sans-serif;
    font-size: 1.2rem;
    letter-spacing: 1px;
    margin-bottom: 0.5rem;
`;

const StyledText = styled(StyledP)`
    font-weight: 700;
    margin-bottom: 1rem;
`;

const StyledSubmitBtn = styled(StyledOutlineButton)`
    border-color: #207070;
    color: #207070;
    margin-bottom: 0.5rem;
`;

const StyledContinueBtn = styled(StyledGreenSoloButton)`
    font-size: 1.2rem;
    border: 2px solid #207070;
`;

const StyledImg = styled.img`
    width: 100%;
`;

const AdminHomepageImageForm = ({ handleImageSubmit, image }) => {
    // set up react-hook-form
    const {
        register,
        handleSubmit,
        formState: { errors },
    } = useForm();
    // set up history
    const history = useHistory();

    return (
        <>
            <StyledUploadForm onSubmit={handleSubmit(handleImageSubmit)}>
                <StyledUploadInput
                    type="file"
                    id="upload"
                    name="upload"
                    data-testid="upload"
                    {...register("upload", {
                        required: "Image is required.",
                    })}
                />
                <StyledSubmitBtn type="submit" name="homepage-submit-btn">
                    Submit
                </StyledSubmitBtn>
                <StyledContinueBtn onClick={() => history.go(0)}>
                    Keep Image
                </StyledContinueBtn>
            </StyledUploadForm>
            <StyledText>Current Image:</StyledText>
            <StyledImg src={image} alt="Homepage" />
        </>
    );
};

export default AdminHomepageImageForm;
