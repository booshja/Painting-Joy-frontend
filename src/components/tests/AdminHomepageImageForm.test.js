import { render } from "@testing-library/react";
import AdminHomepageImageForm from "../AdminHomepageImageForm";

/** Smoke Test */
it("renders without crashing", () => {
    render(
        <AdminHomepageImageForm
            handleImageSubmit={null}
            image="http://lorempixel.com/200/200"
        />
    );
});

/** Snapshot Test */
it("matches snapshot", () => {
    const { asFragment } = render(<AdminHomepageImageForm />);
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders correctly", () => {
    const { getByTestId, getByAltText, getByRole, getByText } = render(
        <AdminHomepageImageForm
            handleImageSubmit={null}
            image="http://lorempixel.com/200/200"
        />
    );

    getByTestId("upload");
    getByRole("button", { name: "Submit" });
    getByRole("button", { name: "Keep Image" });
    getByText("Current Image:");
    getByAltText("Homepage");
});
