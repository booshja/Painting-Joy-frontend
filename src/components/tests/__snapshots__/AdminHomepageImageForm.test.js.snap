// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c4 {
  background-color: #207070;
  font-family: "News Cycle",sans-serif;
  font-weight: 700;
  font-size: 1rem;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  padding: 5px 10px;
  border: none;
  border-radius: 4px;
  color: #ffffff;
}

.c4:hover:not(:disabled) {
  color: #207070;
  background-color: #ffffff;
  border: 2px solid #207070;
  padding: 3px 8px;
  cursor: pointer;
}

.c4:disabled {
  background-color: #dadada;
}

.c2 {
  border-radius: 4px;
  background-color: #ffffff;
  font-family: "News Cycle",sans-serif;
  font-weight: 700;
  font-size: 1.2rem;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  padding: 2.5px 5px;
  border: 2px solid;
}

.c2:hover:not(:disabled) {
  color: #ffffff;
  cursor: pointer;
}

.c2:disabled {
  color: #dadada;
  border: 2px solid #dadada;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 2px solid #207070;
  border-radius: 4px;
  background-color: #ffffff;
  padding: 1rem;
  width: 100%;
  margin-bottom: 1rem;
}

.c1 {
  font-family: "News Cycle",sans-serif;
  font-size: 1.2rem;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  margin-bottom: 0.5rem;
}

.c3 {
  border-color: #207070;
  color: #207070;
  margin-bottom: 0.5rem;
}

.c5 {
  font-size: 1.2rem;
  border: 2px solid #207070;
}

<form
    class="c0"
  >
    <input
      class="c1"
      data-testid="upload"
      id="upload"
      name="upload"
      type="file"
    />
    <button
      class="c2 c3"
      name="homepage-submit-btn"
      type="submit"
    >
      Submit
    </button>
    <button
      class="c4 c5"
    >
      Keep Image
    </button>
  </form>
  .c0 {
  font-size: 1.2rem;
  font-family: "News Cycle",sans-serif;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c1 {
  font-weight: 700;
  margin-bottom: 1rem;
}

<p
    class="c0 c1"
  >
    Current Image:
  </p>
  .c0 {
  width: 100%;
}

<img
    alt="Homepage"
    class="c0"
  />
</DocumentFragment>
`;
